design/authenticators/authn_gcp/authn_gcp_solution_design.md

```diff
@@ -0,0 +1,582 @@
+    
+    * Host annotations implementation  -  Fetch host annotations values, this logic is relevant to GCP, JWT, Azure, K8s
+    
+    3.2 **Validate Source Restrictions** - Each authenticator will have his special logic of which annotation are permitted  (In perfect world this logic should be trigger in load policy stage)
+    
+    * GCP implementation - validate GCP annotations logic (instance-name OR project-id OR service-account-id OR service-account-email)
+    * Azure needs refactor - same Azure logic
+    * K8s needs refactor - same K8S logic
+    * JWT design phase - any key is permitted
+    
+    3.3 **Validate destination Restrictions** - In GCP,JWT,Azure we are validating against JWT token, in K8S against api 
+      
+      3.3.1 **Key Adapter** - Responsible of mapping source key to hist corresponding Destination key value 
+    
+    * GCP implementation - use a key adapter to find the corresponding key to validate
+    * Azure needs refactor - same Azure xms_mirid logic
+    * K8s needs refactor - same K8S logic
+    * JWT design phase - straightforward validation 
+
+#### GCP authenticator flow
+![Authn gcp flow](authn-gcp-flow.png)
+
+
+### Backwards compatibility
+None because this is a new feature.
+
+### Performance
+GCP authenticator performance should conform with our other authenticators with an average call time of 1 second.
+
+### Affected Components
+- Conjur
+
+- DAP
+
+## Security
+* **Future support in Service account key needs to be done very carefully** since there is a concern of impersonate other Conjur hosts, 
+because the Service account key enable you to control the content of the JWT token. 
+The reason it is not a concern in our use case, because we are validating only tokens which supplied by google metadata server, 
+and signed by google private keys which are not expose to the apps in contrast to the Service account key.
+
+## Test Plan
+[//]: # (Fill in the table below to depict the tests that should run to validate your solution)
+[//]: # (You can use this tool to generate a table - https://www.tablesgenerator.com/markdown_tables#)
+
+| **Title** | **Given** | **When** | **Then** | **Comment** |
+|-----------|-----------|----------|----------|-------------|
+|           |           |          |          |             |
+|           |           |          |          |             |
+
+## Logs
+
+### Error Log level
+
+|    | Scenario                                                                | Log message                                                                            | Comment            |
+|--- |-----------------------------------------------------------------------  |----------------------------------------------------------------------------------------|------------------------|
+| 1  | Authenticator is not enabled (in DB/ENV)                                | Authenticator '{0-authenticator-name}' is not enabled                                  | |
```

Comment:
@nessiLahav can you please expand the scenario saying that one called an authenticator that is disabled?

Reply:
Those message are aligned with Azure, at the development stage you will review them again